packages/hawtio/src/plugins/camel/debug/Debug.tsx

```diff
@@ -0,0 +1,782 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import {
+  Card,
+  CardBody,
+  Text,
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+  Button,
+  CardTitle,
+  CardHeader,
+  CardActions,
+  Modal,
+  ModalVariant,
+  Form,
+  FormGroup,
+  Radio,
+  TextInput,
+  Popover,
+  Panel,
+  PanelMain,
+  PanelMainBody,
+  Nav,
+  NavList,
+  NavItem,
+  DrawerHead,
+  DrawerPanelContent,
+  DrawerActions,
+  DrawerCloseButton,
+  Drawer,
+  DrawerContent,
+  DrawerContentBody,
+  Divider,
+} from '@patternfly/react-core'
+import {
+  BanIcon,
+  BarsIcon,
+  ExclamationCircleIcon,
+  HelpIcon,
+  LongArrowAltDownIcon,
+  MinusIcon,
+  PlayIcon,
+  PlusCircleIcon,
+  PlusIcon,
+  TimesCircleIcon,
+} from '@patternfly/react-icons'
+import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table'
+import { ConditionalBreakpoint, debugService as ds, MessageData } from './debug-service'
+import * as ccs from '../camel-content-service'
+import { MBeanNode } from '@hawtiosrc/plugins/shared'
+import { isArray } from '@hawtiosrc/util/objects'
+import { RouteDiagram } from '../route-diagram/RouteDiagram'
+import './Debug.css'
+import { Annotation, RouteDiagramContext, useRouteDiagramContext } from '../route-diagram/route-diagram-context'
+import { CamelNodeData } from '../route-diagram/visualization-service'
+import { IResponse } from 'jolokia.js'
+import { log } from '../globals'
+import { parseXML } from '@hawtiosrc/util/xml'
+
+export const Debug: React.FunctionComponent = () => {
```

Comment:
This component seems to have a bit too many states in a single component. Perhaps we can split it into smaller ones so each can handle a smaller set of the states.

Reply:
What I suggested was to split the single `Debug` react component into smaller parts, like `Debug` -> `DebugToolbar`, `DebugPanel` (if that makes sense). When you actually do so, you'll be surprised to find how React can manage states in smaller chunks in a more maintainable way.

But it's not a big deal anyway, we can always polish code later.